examples/vision/image_classification_with_vision_transformer.py

```diff
@@ -0,0 +1,294 @@
+"""
+Title: Image Classification with Vision Transformer
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/01/30
+Last modified: 2021/01/30
+Description: Implementing Vision Transformer (ViT) model for image classification.
+"""
+
+"""
+## Introduction
+
+This example implements the [Vision Transformer (ViT)](https://arxiv.org/abs/2010.11929)
+model by Alexey Dosovitskiy et al. for image classification on the CIFAR-100 dataset.
+The ViT applies the Transformer architecture with self-attentions on sequences of
+image patches without using convolutional networks.
+
+The example requires TensorFlow 2.4 or higher and
+[TensorFlow Addons](https://www.tensorflow.org/addons/overview),
```

Comment:
Specify that TFA is used for the AdamW optimizer only

Reply:
I actually now using just Adam, so I removed the need for tfa